app/src/validators/permission.js

```diff
@@ -0,0 +1,60 @@
+const { validate, Joi } = require('express-validation');
+const { uuidv4, uuidv4MultiModel, permCodeMultiModel } = require('./common');
+const { Permissions } = require('../components/constants');
+
+const schema = {
+  searchPermissions: {
+    query: Joi.object({
+      userId: uuidv4MultiModel,
+      objId: uuidv4MultiModel,
+      permCode: permCodeMultiModel
+    }).min(1)
+  },
+
+  listPermissions: {
+    query: Joi.object({
+      userId: uuidv4MultiModel,
+      objId: uuidv4MultiModel,
+      permCode: permCodeMultiModel
+    })
+  },
+
+  addPermissions: {
+    params: Joi.object({
+      objId: uuidv4
```

Comment:
not sure .. do you need to mark this as required?

Reply:
Jeremy and I discussed marking the params required, but as a param of the route they are already explicitly required to call it. It would also require a duplicate object of uuidv4 with the required tag to be created or the unit testing will fail. So we scrapped the idea.